src/Umbraco.Configuration/Models/ContentSettings.cs

```diff
@@ -0,0 +1,89 @@
+        public IEnumerable<IImagingAutoFillUploadField> ImageAutoFillProperties => _configuration.GetValue<ImagingAutoFillUploadField[]>("Umbraco:CMS:Core:Content:Imaging:autoFillImageProperties") ?? DefaultImagingAutoFillUploadField;
+
+
+        public bool ResolveUrlsFromTextString => _configuration.GetValue<bool?>("Umbraco:CMS:Content:ResolveUrlsFromTextString") ?? false;
+        public IEnumerable<IContentErrorPage> Error404Collection => _configuration
+            .GetSection("Umbraco:CMS:Content:Errors:Error404")
+            .GetChildren()
+            .Select(x=> new ContentErrorPage(x));
+        public string PreviewBadge =>  _configuration.GetValue<string>("Umbraco:CMS:Content:PreviewBadge") ?? DefaultPreviewBadge;
+        public MacroErrorBehaviour MacroErrorBehaviour => _configuration.GetValue<MacroErrorBehaviour?>("Umbraco:CMS:Content:MacroErrors") ?? MacroErrorBehaviour.Inline;
+        public IEnumerable<string> DisallowedUploadFiles => _configuration.GetValue<string[]>("Umbraco:CMS:Content:DisallowedUploadFiles") ?? new[] {"ashx", "aspx", "ascx", "config", "cshtml", "vbhtml", "asmx", "air", "axd"};
+        public IEnumerable<string> AllowedUploadFiles => _configuration.GetValue<string[]>("Umbraco:CMS:Content:AllowedUploadFiles") ?? Array.Empty<string>() ;
+        public bool ShowDeprecatedPropertyEditors =>  _configuration.GetValue<bool?>("Umbraco:CMS:Content:ShowDeprecatedPropertyEditors") ?? false;
+        public string LoginBackgroundImage =>  _configuration.GetValue<string>("Umbraco:CMS:Content:LoginBackgroundImage") ?? string.Empty;
+
+        private class ContentErrorPage : IContentErrorPage
+        {
+            public ContentErrorPage(IConfigurationSection configurationSection)
+            {
+                Culture = configurationSection.Key;
+
+                var value = configurationSection.Value;
+
+                if(int.TryParse(value, out var contentId))
+                {
+                    HasContentId = true;
+                    ContentId = contentId;
+                }
+                else if(Guid.TryParse(value, out var contentKey))
+                {
+                    HasContentKey = true;
+                    ContentKey = contentKey;
+                }
+                else
+                {
+                    ContentXPath = value;
+                }
+            }
+
+            public int ContentId { get; set; }
+            public Guid ContentKey { get; set;}
+            public string ContentXPath { get; set;}
+            public bool HasContentId { get; set;}
+            public bool HasContentKey { get; set;}
+            public string Culture { get; set; }
+        }
+
+        private class ImagingAutoFillUploadField : IImagingAutoFillUploadField
+        {
+            public string Alias { get; set; }
```

Comment:
Just checking for my own understanding - the decision for both these classes ImagingAutoFillUploadField and ContentErrorPage to be nested classes within the Internal ContentSettings class, is this to control the usage to be internal to ContentSettings?

Reply:
The only reason is because they are only used by `ContentSettings` and everything people should care about is available on the interface..